akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala

```diff
@@ -120,6 +120,14 @@ sealed trait HttpMessage extends jm.HttpMessage {
     }
   }
 
+  /** Returns all the headers of the given type **/
+  def headers[T >: Null <: jm.HttpHeader: ClassTag]: immutable.Seq[T] = {
```

Comment:
Here's a simpler implementation:

```
  /** Returns all the headers of the given type **/
  def headers[T <: jm.HttpHeader: ClassTag]: immutable.Seq[T] =
    headers.collect { case h: T ⇒ h }
```

The `>: Null` type bound is only needed for `header` because of its (arguably useful) usage of `OptionVal`.

Reply:
Aha!! Thanks for that, I wasn't sure why the need for `>: Null`. I'll fix it later :)